include/bluetooth/mesh/net_db.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2019 Tobias Svehagen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_MESH_NET_DB_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_MESH_NET_DB_H_
+
+#if defined(CONFIG_BT_MESH_NET_DB)
+#define NODE_COUNT    CONFIG_BT_MESH_NET_DB_NODE_COUNT
+#define SUBNET_COUNT  CONFIG_BT_MESH_NET_DB_SUBNET_COUNT
+#define APP_KEY_COUNT CONFIG_BT_MESH_NET_DB_APP_KEY_COUNT
+#else
+#define NODE_COUNT    0
+#define SUBNET_COUNT  0
+#define APP_KEY_COUNT 0
+#endif
+
+struct bt_mesh_db_node {
+	u16_t addr;
+	u16_t net_idx; /* XXX: is this needed? */
+	u8_t  dev_key[16];
+	u8_t  num_elem;
+};
```

Comment:
Should we not include something to track the UUID of the node? 
`u8_t  uuid[16];         /* node uuid */`

There may be a need to ensure that the addr and the UUID match. Then un-provision and reset and node if duplicate addr or if the same beacon/node attempts to reconnect to network then it would be assigned the original assigned addr.

Reply:
Yup that is a good idea and it will be included in the next update tomorrow hopefully.

Reply:
You may find it interesting to have a look at;

(https://docs.espressif.com/projects/esp-idf/en/latest/api-guides/esp-ble-mesh/ble-mesh-architecture.html)

It is an good source for bench-marking ideas

Reply:
@WilliamGFish that's indeed a quite nice overview of Mesh, and particularly relevant since the Espressif implementation is a fork of the Zephyr Mesh stack.